### ISSUE TITLE
Why am I able to use EventGridEvent in isolated Azure Functions?

### ISSUE THREAD

**Issue by elias-rod** [Azure/Azure-Functions#2331]
I'm able to execute an EventGrid triggered Azure Function in isolated mode such as:

    [Function(nameof(PocEventGridTriggerAsync))]
    public static async Task PocEventGridTriggerAsync(
        [EventGridTrigger] EventGridEvent eventGridEvent,
        FunctionContext executionContext)

But in the official documentation it gives [an example][1] and also [states][2] that:

> Because .NET isolated projects run in a separate worker process, bindings can't take advantage of rich binding classes, such as ICollector<T>, IAsyncCollector<T>, and CloudBlockBlob. There's also no direct support for types inherited from underlying service SDKs, such as DocumentClient and BrokeredMessage. Instead, bindings rely on strings, arrays, and serializable types, such as plain old class objects (POCOs).

Am I missing something?

  [1]: https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-event-grid-trigger?tabs=isolated-process%2Cextensionv3&pivots=programming-language-csharp#example
  [2]: https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide#:~:text=Because%20.NET%20isolated%20projects,old%20class%20objects%20(POCOs).

**Comment by bhagyshricompany** [comment]
Hi @elias-rod It's great that you were able to successfully execute an EventGrid-triggered Azure Function in isolated mode.

Regarding the official documentation, it is correct that .NET isolated projects run in a separate worker process and therefore cannot take advantage of rich binding classes such as ICollector, IAsyncCollector, and CloudBlockBlob. Instead, these bindings rely on strings, arrays, and serializable types such as plain old class objects (POCOs).

The example you provided, where you are using an EventGridTrigger, is an exception to this. EventGridTrigger is a built-in binding that is specifically designed to work with isolated functions. It allows you to receive events from Azure Event Grid and use the EventGridEvent class to access the data contained in the event. This is why it is possible to use the EventGridEvent class directly in your function code.

However, other bindings such as BlobTrigger, QueueTrigger, and CosmosDBTrigger are not designed to work with isolated functions, and you will have to use their corresponding string or POCO representation instead.

In summary, EventGridTrigger is an exception that allows to use rich binding classes in isolated functions, but for other bindings, it is recommended to use strings, arrays, and serializable types like POCOs.

**Comment by bhagyshricompany** [comment]
Yes, it's great that you are able to execute an EventGrid-triggered Azure Function in isolated mode. As I mentioned before, EventGridTrigger is an exception that allows you to use rich binding classes in isolated functions, which means you can use the EventGridEvent class directly in your function code.

Other examples of exceptions that allow you to use rich binding classes in isolated functions include:

HttpTrigger: which allows you to use the HttpRequest and HttpResponse classes directly in your function code
TimerTrigger: which allows you to use the TimerInfo class directly in your function code.
BlobTrigger: which allows you to use the CloudBlockBlob class directly in your function code
QueueTrigger: which allows you to use the CloudQueueMessage class directly in your function code
It's worth noting that even though there are these exceptions, it's still recommended to use the string or POCO representation for other bindings in isolated functions, as it provides a more consistent behavior across all hosting models.

You are correct, part of the documentation should be updated to reflect these exceptions, and I apologize for any confusion caused. I'd suggest checking the official documentation from time to time for updates.

**Comment by elias-rod** [comment]
Something does not seem right with your answers
![image](https://user-images.githubusercontent.com/12984787/214385557-600660a4-eb9f-4576-9d9a-0d47a224150b.png)

**Comment by elias-rod** [comment]
👌

**Comment by bhagyshricompany** [comment]
@fabiocav pls comment and validate

**Comment by bhagyshricompany** [comment]
@elias-rod  pls share all repro steps.
Thanks
